libbpf-rs/src/object.rs

```diff
@@ -77,43 +80,23 @@ impl ObjectBuilder {
         let path_c = util::str_to_cstring(path_str)?;
         let path_ptr = path_c.as_ptr();
 
-        // Convert name to a C style pointer
-        //
-        // NB: we must hold onto a CString otherwise our pointer dangles
-        let name = util::str_to_cstring(&self.name)?;
-        let name_ptr = if !self.name.is_empty() {
```

Comment:
I am wondering why we had this case in there. Does `libbpf` not like an empty string as name? It seems your logic changed semantics, where we now set `name` to `NULL` only if not set at all, but not when it's an empty string. Should we add a test for that (e.g., in `test_object_name`)?